standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1360,19 +1384,49 @@ public void create_database(final Database db)
         }
         Deadline.checkTimeout();
       }
+      Database db = new Database(createDatabaseRequest.getDatabaseName(), createDatabaseRequest.getDescription(),
+              createDatabaseRequest.getLocationUri() ,createDatabaseRequest.getParameters());
+      if (createDatabaseRequest.isSetPrivileges()) {
+        db.setPrivileges(createDatabaseRequest.getPrivileges());
+      }
+      if (createDatabaseRequest.isSetOwnerName()) {
+        db.setOwnerName(createDatabaseRequest.getOwnerName());
+      }
+      if (createDatabaseRequest.isSetOwnerType()) {
+        db.setOwnerType(createDatabaseRequest.getOwnerType());
+      }
+      db.setCatalogName(createDatabaseRequest.getCatalogName());
+      if (createDatabaseRequest.isSetCreateTime()) {
+        db.setCreateTime(createDatabaseRequest.getCreateTime());
+      }
+      if (createDatabaseRequest.isSetManagedLocationUri()) {
+        db.setManagedLocationUri(createDatabaseRequest.getManagedLocationUri());
+      }
+      if (createDatabaseRequest.isSetType()) {
+        db.setType(createDatabaseRequest.getType());
+      }
+      if (createDatabaseRequest.isSetDataConnectorName()) {
+        db.setConnector_name(createDatabaseRequest.getDataConnectorName());
+      }
+      if (createDatabaseRequest.isSetRemote_dbname()) {
+        db.setRemote_dbname(createDatabaseRequest.getRemote_dbname());
+      }
       create_database_core(getMS(), db);
+      createDatabaseRequest.setLocationUri(db.getLocationUri());
```

Comment:
Why do we change the `createDatabaseRequest` after `create_database_core`

Reply:
The same reasoning I gave for modification of `db` object after create_database_req()

Reply:
No one would change the inner variables of `createDatabaseRequest`, right?